src/main/java/tina/Parser.java

```diff
@@ -0,0 +1,146 @@
+package tina;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import tina.task.Deadline;
+import tina.task.Event;
+import tina.task.Task;
+import tina.task.Todo;
+
+
+/**
+ * The <code>Parser</code> class handles the parsing of user input commands and task data.
+ * It provides methods to interpret commands, convert strings into <code>Task</code> objects,
+ * and format dates.
+ */
+public class Parser {
+
+    /**
+     * Parses a task creation command and adds the task to the provided <code>TaskList</code>.
+     *
+     * @param input The user input command for creating a task.
+     * @param tasks The <code>TaskList</code> to which the parsed task will be added.
+     * @return A message confirming the task has been added to the <code>TaskList</code>.
+     * @throws TinaException if the command is invalid or the description is missing.
+     */
+    private static String parseTask(String input, TaskList tasks) throws TinaException {
+        try {
+            if (input.startsWith("todo")) {
+                String des = input.substring(5);
+                return tasks.addTask(new Todo(des));
+            } else if (input.startsWith("deadline")) {
+                int byIdx = input.indexOf("/by");
+                String des = input.substring(9, byIdx - 1);
+                String end = input.substring(byIdx + 4);
+                return tasks.addTask(new Deadline(des, end));
+            } else if (input.startsWith("event")) {
+                int fromIdx = input.indexOf("/from");
+                int toIdx = input.indexOf("/to");
+                String des = input.substring(6, fromIdx - 1);
+                String start = input.substring(fromIdx + 6, toIdx - 1);
+                String end = input.substring(toIdx + 4);
+                return tasks.addTask(new Event(des, start, end));
+            } else {
+                throw new TinaException("I have no idea what that means");
+            }
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new TinaException("Where is your description?");
+        }
+    }
```

Comment:
Excellent work! The names of the methods are verbs and relevant to the task it is performing